linera-chain/src/manager.rs

```diff
@@ -369,10 +369,10 @@ impl ChainManager {
         &self,
         certificate: &ValidatedBlockCertificate,
     ) -> Result<Outcome, ChainError> {
-        let new_block = &certificate.executed_block().block;
+        let new_block = certificate.block();
         let new_round = certificate.round;
         if let Some(Vote { value, round, .. }) = &self.confirmed_vote {
-            if value.inner().executed_block().block == *new_block && *round == new_round {
+            if value.inner().block() == new_block && *round == new_round {
```

Comment:
TODO: Double-check. Previously we were comparing the `Proposal` and now we compare the new `Block` type. It means we also compare the execution results now.

Reply:
Either way is okay; we could even just compare the round: There can't be more than one validated block certificate per round anyway, unless there is no quorum of honest validators.